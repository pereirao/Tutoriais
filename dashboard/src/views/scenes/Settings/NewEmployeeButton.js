// @flow

import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import I18n from '../../../localization';
import NewEmployeeModal from './NewEmployeeModal';
import MBButton from '../../elements/MBButton';
import MBIcon from '../../elements/MBIcon';
import { employee_actions } from '../../../business/employee';

type NewEmployeeButtonState = {is_modal_open: boolean};
export class NewEmployeeButton extends PureComponent {
  state: NewEmployeeButtonState

  constructor(props){
    super(props);
    this.state = { is_modal_open: false };
  }

  toggleModalVisibility = () => {
    this.props.resetForm();
    this.setState({ is_modal_open: !this.state.is_modal_open });
  }

  render(){
    return (
      <div>
        <MBButton
          title={I18n.t('accessibility.title.add_employee')}
          button_type="link"
          size="large"
          onClick={this.toggleModalVisibility}>
          <MBIcon icon="add" color="mb_red" size="small" />
          <div>{I18n.t('ui.button.new_employee')}</div>
        </MBButton>
        <NewEmployeeModal hidden={!this.state.is_modal_open} close={this.toggleModalVisibility} />
      </div>
    );
  }
}

const NewEmployeeFormDTP = {
  resetForm: employee_actions.resetNewEmployeeForm
};
const NewEmployeeButtonContainer = connect(null, NewEmployeeFormDTP)(NewEmployeeButton);

export default NewEmployeeButtonContainer;
