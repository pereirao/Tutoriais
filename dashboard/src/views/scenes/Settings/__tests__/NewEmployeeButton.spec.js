import React from 'react';
// NOTE: the named export isn't a connected component
import { NewEmployeeButton } from '../NewEmployeeButton';

it('renders correctly', () => {
  expect(shallow(<NewEmployeeButton />)).toMatchSnapshot();
});
