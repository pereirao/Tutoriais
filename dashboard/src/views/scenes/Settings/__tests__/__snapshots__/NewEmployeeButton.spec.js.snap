// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <MBButton
    button_type="link"
    onClick={[Function]}
    size="large"
    title="Add a New Employee or Driver"
  >
    <MBIcon
      color="mb_red"
      icon="add"
      size="small"
    />
    <div>
      New Employee/Driver
    </div>
  </MBButton>
  <NewEmployeeModal
    close={[Function]}
    hidden={true}
  />
</div>
`;
